Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -231,6 +238,53 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
     self.showErrorBannerWithMessage = Signal
       .combineLatest(validateCheckoutExistingCard.errors(), validateCheckoutNewCard.errors())
       .map { _ in Strings.Something_went_wrong_please_try_again() }
+
+    let paymentAuthorizationData: Signal<PaymentAuthorizationData, Never> = self.configureWithDataProperty
+      .signal
+      .skipNil()
+      .map { (data: PostCampaignCheckoutData) -> PaymentAuthorizationData in
+        let reward = data.rewards.first! // What happens if you select no reward?
```

Comment:
Nit: While I don't think it's possible for there to be no reward (selecting "no reward" means we create a special "no reward" reward), can we fail gracefully if this were to happen instead of crashing the program?

Reply:
Yeah technically there is a special reward with its own rewardID that we should be aware of.